src/main/java/com/hierynomus/smbj/share/DiskShare.java

```diff
@@ -58,6 +60,25 @@
 
     private static final Logger logger = LoggerFactory.getLogger(DiskShare.class);
 
+    protected ShareInfo shareInfo = null;
+    protected long attrExpiration = 0L;
+    protected static final long attrExpirationPeriod;
+    
+    protected static final long DEFAULT_ATTR_EXPIRATION_PERIOD = 5000;
+    
+    static {
```

Comment:
I don't think it should be the responsibility of this library to keep a (hard to tune) cache of the retrieved info. This typically belongs to the calling application.